File: app/src/main/java/org/ole/planet/myplanet/base/BaseDialogFragment.java
package org.ole.planet.myplanet.base;

import android.os.Bundle;
import androidx.annotation.Nullable;
import androidx.fragment.app.DialogFragment;


public abstract class BaseDialogFragment extends DialogFragment {
  public  String id;
    @Override
    public void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setStyle(DialogFragment.STYLE_NO_TITLE, android.R.style.Theme_Holo_Light_Dialog_NoActionBar_MinWidth);
        if (getArguments() != null) {
            id = getArguments().getString(getKey());
        }
    }

     protected abstract String getKey();

}

File: app/src/main/java/org/ole/planet/myplanet/base/BaseDialogFragment.kt
package org.ole.planet.myplanet.base

import org.ole.planet.myplanet.R
import android.os.Bundle
import androidx.fragment.app.DialogFragment

abstract class BaseDialogFragment : DialogFragment() {
    var id: String? = null
    var teamId: String? = null
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setStyle(STYLE_NO_TITLE, R.style.AppTheme_Dialog_NoActionBar_MinWidth)
        if (arguments != null) {
            id = requireArguments().getString(key)
            teamId = requireArguments().getString("teamId")
        }
    }

    protected abstract val key: String?
}
